kimchi/src/prover.rs

```diff
@@ -110,8 +110,8 @@ where
             // padding
             w.extend(std::iter::repeat(ScalarField::<G>::zero()).take(length_padding));
 
-            // zk-rows
-            for row in w.iter_mut().rev().take(ZK_ROWS as usize) {
+            // padding for eval-rows = zk-rows + acc-rows
```

Comment:
this is not padding, this is the zero-knowledge rows

Reply:
Can't we use `PERM_FINAL_CONSTRAINTS` here since it is also equal to `EVALS + 1` with the new approach?

Reply:
I guess there's a number of different polynomials/arguments that we zk-ify:

- witness (here)
- permutation
- lookup

and they all rely on the global number of evaluations (which is 2, but we can set it to 3 just to make sure we have enough random values)

Reply:
(btw don't hesitate to hit me up on twitter if you want to talk more about this)